### PR TITLE
feat(anta.tests): Add LANZ, PTP and GreenT

### PR DESCRIPTION
**Pull request by BarikiVarshitha**
# Description

Add new tests for LANZ , GreenT and PTP.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tests/units/anta_tests/test_lanz.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from anta.tests.lanz import VerifyLANZ
+from tests.lib.anta import test  # noqa: F401; pylint: disable=W0611
+
+DATA: list[dict[str, Any]] = [
+    {
+        "name": "success",
+        "test": VerifyLANZ,
+        "eos_data": [{"lanzEnabled": True}],
+        "inputs": None,
+        "expected": {"result": "success"},
```
Comment (mtache) [inline tests/units/anta_tests/test_lanz.py:18]:
```suggestion
        "expected": {"result": "success", "messages": ["LANZ is enabled"]},
```

File: tests/units/anta_tests/test_lanz.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    },
+    {
+        "name": "failure",
+        "test": VerifyLANZ,
+        "eos_data": [{"lanzEnabled": False}],
+        "inputs": None,
+        "expected": {"result": "failure"},
```
Comment (mtache) [inline tests/units/anta_tests/test_lanz.py:25]:
```suggestion
        "expected": {"result": "failure", "messages": ["LANZ is not enabled"]},
```

File: anta/tests/ptp.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""
+Test functions related to PTP (Precision Time Protocol) in EOS
+"""
+# Mypy does not understand AntaTest.Input typing
+# mypy: disable-error-code=attr-defined
```
Comment (mtache) [inline anta/tests/ptp.py:8]:
```suggestion
```

File: anta/tests/lanz.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""
+Test functions related to LANZ
+"""
+# Mypy does not understand AntaTest.Input typing
+# mypy: disable-error-code=attr-defined
```
Comment (mtache) [inline anta/tests/lanz.py:8]:
```suggestion
# Mypy does not understand AntaTest.Input typing
# mypy: disable-error-code=attr-defined
```

File: anta/tests/greent.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""
+Test functions related to GreenT (Postcard Telemetry) in EOS
+"""
+# Mypy does not understand AntaTest.Input typing
+# mypy: disable-error-code=attr-defined
```
Comment (mtache) [inline anta/tests/greent.py:8]:
```suggestion
```

File: anta/tests/greent.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    """
+    Verifies whether GreenT policy is created.
+
+    Expected Results:
+        * success: if there exists any policy other than "default" policy.
+        * failure: if no policy is created.
+    """
+
+    name = "VerifyGreenT"
+    description = "Verifies if the greent."
```
Comment (mtache) [inline anta/tests/greent.py:48]:
```suggestion
    description = "Verifies whether GreenT policy is created."
```

**Review by mtache — CHANGES_REQUESTED** [review]
Great PR !

File: tests/units/anta_tests/test_lanz.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+from anta.tests.lanz import VerifyLANZ
+from tests.lib.anta import test  # noqa: F401; pylint: disable=W0611
+
+DATA: list[dict[str, Any]] = [
+    {
+        "name": "success",
+        "test": VerifyLANZ,
+        "eos_data": [{"lanzEnabled": True}],
+        "inputs": None,
+        "expected": {"result": "success", "message": ["LANZ is enabled"]},
```
Comment (MaheshGSLAB) [inline tests/units/anta_tests/test_lanz.py:18]: message is not needed

Comment (MaheshGSLAB) [inline tests/units/anta_tests/test_lanz.py:18]:
```suggestion
        "expected": {"result": "success", "messages": ["LANZ is enabled"]},
```

**Review by mtache — APPROVED** [review]

**Pull request merged**